src/motion-filter.c

```diff
@@ -358,7 +359,7 @@ static bool init_hotkey(void *data)
 	obs_scene_t *scene = obs_scene_from_source(source);
 	filter->hotkey_init = true;
 
-	if (!scene)
+	if (!scene || source->context.private)
```

Comment:
if (!scene || source->context.private){
  // do obs_hotkey_pair_register_frontend() or other workaround way to make it work in studio mode
}